flang/lib/Lower/IO.cpp

```diff
@@ -225,11 +225,7 @@ static mlir::FuncOp getOutputFunc(mlir::Location loc,
   // Use descriptors for arrays
   if (auto refTy = type.dyn_cast<fir::ReferenceType>())
     type = refTy.getEleTy();
-  if (type.isa<fir::SequenceType>())
-    return getIORuntimeFunc<mkIOKey(OutputDescriptor)>(loc, builder);
-  // Any unaccounted for types are to be handled here.
-  mlir::emitError(loc, "output for entity type ") << type << " not implemented";
-  TODO(loc, "derived type IO");
```

Comment:
Same, here the TODO is still not yet done, this will just crash later.

Reply:
I think "later" here is fine and caught with a message about no code to lower to LLVM. There is no reason then to catch this as early as possible. Removing the TODO exposes other problems, like the TODO in codegen.